src/dmqnode/storage/Ring.d

```diff
@@ -315,18 +315,21 @@ public class RingNode : StorageChannels
 
         override public typeof(this) pop ( ref char[] value )
         {
-            void[] allocValue ( size_t n )
+            scope allocValue = delegate void[] ( size_t n )
             {
+                if (value.length < n)
+                    enableStomping(value);
```

Comment:
That looks weird. You `enableStomping` in the event you extend the buffer, but that is AFAIK useless because extending the buffer == reallocation (most of the time).
I would expect a single `enableStomping` call after the `length` assignment.